clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -183,22 +183,33 @@ public RecordAppendResult append(TopicPartition tp,
             ByteBuffer buffer = free.allocate(size, maxTimeToBlock);
             synchronized (dq) {
                 // Need to check if producer is closed again after grabbing the dequeue lock.
-                if (closed)
+                if (closed) {
+                    free.deallocate(buffer);
```

Comment:
It seems OK to me not to free this buffer if the accumulator is closed since we no longer need this BufferPool. Is there any consequence of not calling `free.deallocate(buffer)` here?